engine/src/main/java/org/camunda/bpm/engine/impl/db/EnginePersistenceLogger.java

```diff
@@ -314,9 +315,19 @@ public ProcessEngineException illegalValueForUserIdException(String id, String e
       exceptionMessage("028", "Illegal value '{}' for userId for GLOBAL authorization. Must be '{}'", id, expected));
   }
 
-  public AuthorizationException requiredCamundaAdminException() {
-    return new AuthorizationException(
-      exceptionMessage("029", "Required admin authenticated group or user."));
+  public AuthorizationException requiredCamundaAdminOrPermissionException(List<MissingAuthorization> missingAuthorizations) {
+    String exceptionCode;
```

Comment:
How about we initialize the previous default case and only change the code when we have missing authorizations?
```suggestion
    String exceptionCode = "029";
```